module/VuFind/src/VuFind/Service/MarkdownFactory.php

```diff
@@ -63,29 +77,41 @@ class MarkdownFactory implements FactoryInterface
     public function __invoke(
         ContainerInterface $container, $requestedName, array $options = null
     ) {
-        $config = $container->get(\VuFind\Config\PluginManager::class)
-            ->get('markdown')->Markdown;
-        return new GithubFlavoredMarkdownConverter(
-            [
-                'html_input' => $config->html_input ?? 'strip',
-                'allow_unsafe_links' => $config->allow_unsafe_links ?? false,
-                'enable_em' => $config->enable_em ?? true,
-                'enable_strong' => $config->enable_strong ?? true,
-                'use_asterisk' => $config->use_asterisk ?? true,
-                'use_underscore' => $config->use_underscore ?? true,
-                'unordered_list_markers' => isset($config->unordered_list_markers)
-                    && $config->unordered_list_markers instanceof \ArrayAccess
-                        ? $config->unordered_list_markers->toArray()
-                        : ['-', '*', '+'],
-                'max_nesting_level' => $config->max_nesting_level ?? \INF,
-                'renderer' => [
-                    'block_separator'
-                        => $config->renderer['block_separator'] ?? "\n",
-                    'inner_separator'
-                        => $config->renderer['inner_separator'] ?? "\n",
-                    'soft_break' => $config->renderer['soft_break'] ?? "\n",
-                ],
-            ]
-        );
+        $markdownConfig = $container->get(\VuFind\Config\PluginManager::class)
+            ->get('markdown');
+        $markdownSection = $markdownConfig->Markdown;
+        $environment = Environment::createCommonMarkEnvironment();
+        $environment->addExtension(new GithubFlavoredMarkdownExtension());
+        $config = [
+            'html_input' => $config->html_input ?? 'strip',
+            'allow_unsafe_links' => $config->allow_unsafe_links ?? false,
+            'enable_em' => $config->enable_em ?? true,
+            'enable_strong' => $config->enable_strong ?? true,
+            'use_asterisk' => $config->use_asterisk ?? true,
+            'use_underscore' => $config->use_underscore ?? true,
+            'unordered_list_markers' => isset($config->unordered_list_markers)
+                && $config->unordered_list_markers instanceof \ArrayAccess
+                    ? $config->unordered_list_markers->toArray()
+                    : ['-', '*', '+'],
+            'max_nesting_level' => $config->max_nesting_level ?? \INF,
+            'renderer' => [
+                'block_separator'
+                    => $config->renderer['block_separator'] ?? "\n",
+                'inner_separator'
+                    => $config->renderer['inner_separator'] ?? "\n",
+                'soft_break' => $config->renderer['soft_break'] ?? "\n",
+            ],
+        ];
+        $extensions = isset($markdownSection->extensions)
+            ? explode(',', $markdownSection->extensions) : [];
```

Comment:
It might be a good idea to `array_map('trim', explode(...))` to be more tolerant of whitespace in the config file (but maybe it's not necessary if the library is already whitespace-tolerant).

Reply:
done